File: rules/go/weak-hashes.go
// Test fixtures for pqc-semgrep-rules / Go / classically-broken hashes.
package main

import (
	"crypto/md5"
	"crypto/sha1"
	"crypto/sha256"
)

func vulnerableHashes() {
	// ruleid: pqc.go.md5
	_ = md5.New()
	// ruleid: pqc.go.md5
	_ = md5.Sum([]byte("x"))

	// ruleid: pqc.go.sha1
	_ = sha1.New()
	// ruleid: pqc.go.sha1
	_ = sha1.Sum([]byte("y"))
}

func safeHashes() {
	// ok: pqc.go.md5
	_ = sha256.New()
}

File: rules/javascript/weak-hashes.js
// Test fixtures for pqc-semgrep-rules / JavaScript / classically-broken hashes.
const crypto = require('crypto');

function vulnerable() {
    // ---- MD5 ----
    // ruleid: pqc.javascript.md5
    crypto.createHash('md5').update('x').digest('hex');
    // ruleid: pqc.javascript.md5
    crypto.createHash('MD5').update('x').digest('hex');

    // ---- SHA-1 ----
    // ruleid: pqc.javascript.sha1
    crypto.createHash('sha1').update('y').digest('hex');
    // ruleid: pqc.javascript.sha1
    crypto.createHash('SHA1').update('y').digest('hex');
}

function safe() {
    // ok: pqc.javascript.md5
    crypto.createHash('sha256').update('safe').digest('hex');
}
